ICU-22947 Update user guide for MF2 (#3241)
unicode-org/icu: docs/userguide/format_parse/messages/mf2.md

You should be able to use your preferred IDE (Eclipse, IntelliJ, Visual Studio Code, more), use a different build system, etc.

### ICU4C

Documentation for ICU4C in MF2 is being developed at [messageformat.dev](https://messageformat.dev/docs/integration/cpp/),
which also contains [MF2 syntax documentation](https://messageformat.dev/docs/quick-start/)

> and an interactive [playground](https://messageformat.dev/playground/) for testing messages.

Reviewer: I'm not familiar with this site, is it from MFv2 workign group? Is it implied that that content should be merged back here into this page?

Reviewer 2: That page's quickstart content looks great, and the site overall is wonderful too. But it's not currently a part of Unicode. @lucacasonato Would you be willing to upstream that content to the ICU User Guide? What are your plans?

I'm guessing @catamorphism wants to provide proper documentation for his ICU4C implementation work, and that overlaps with @lucacasonato's existing great work there.

Regardless, the current text in the PR makes it sound like the canonical documentation for using ICU4C MF2 is available externally. If we can upstream the content to the ICU User Guide, then I think it might be nice to link to messageformat.dev as the source for the contribution, if that's not too out of place for ICU documentation to do. Otherwise we consider rewording to make it clear that it's not the canonical source of docs.

messageformat.dev shows how much better we can aspire for Unicode docs to be, given how much benefit all of the projects & data & specs like this provide to people.

Reviewer 3: @echeran Upstreaming is certainly an option, however the content would lose a lot of value if it was not accompanied by its interactive examples - can we just move the entire site into Unicode control and keep the content and interactive examples on the site as is (maybe change domain to a Unicode one?). Linking to it from the ICU docs should not be a problem then.

Author: I just don't want to duplicate the content across two different web sites. I've been contributing my documentation to messageformat.dev so far, but if @lucacasonato wants to move the site under Unicode control, that would solve both problems.

Reviewer 2: @lucacasonato @catamorphism I like the idea of upstreaming the website to Unicode for all of the reasons you mentioned. 

@srl295 does that work? would it be possible to migrate any relevant git repos to the `unicode-org` Github organization? could we create a subdomain like `messageformat.unicode.org` to host the site?

Reviewer: Theoretically yes- a question though would it make sense to just deploy it from the MFWG site?

Reviewer 3: Could do that. But the mfwg repo has a completely different model when it comes to contributions. I for example would still like to have push access to be able to continue to develop the site.

Reviewer: @lucacasonato I'm working on getting the meeting together. This week is Unicode's [UTW](https://www.unicode.org/events/utw/) conference, so might be challenging time wise. 

Who on this thread needs to be included besides @lucacasonato  @echeran @catamorphism ?

> ### ICU4C
> 
> Documentation for ICU4C in MF2 is being developed at [messageformat.dev](https://messageformat.dev/docs/integration/cpp/),
> which also contains [MF2 syntax documentation](https://messageformat.dev/docs/quick-start/)
> and an interactive [playground](https://messageformat.dev/playground/) for testing messages.

Reviewer: For now, can we just note this as a helpful link:

```suggestion
### ICU4C

Some helpful documentation for ICU4C in MF2 is being developed at [messageformat.dev](https://messageformat.dev/docs/integration/cpp/),
as well an unofficial [MF2 syntax documentation](https://messageformat.dev/docs/quick-start/)
and an interactive [playground](https://messageformat.dev/playground/) for testing messages.
```
